pkg/util/cluster/cluster.go

```diff
@@ -341,19 +340,15 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 	return nil
 }
 
-func (c *Cluster) generateSubnets() (vnetPrefix string, masterSubnet string, workerSubnet string) {
+func generateSubnets() (vnetPrefix string, masterSubnet string, workerSubnet string) {
 	// pick a random 23 in range [10.3.0.0, 10.127.255.0]
 	// 10.0.0.0/16 is used by dev-vnet to host CI
 	// 10.1.0.0/24 is used by rp-vnet to host Proxy VM
 	// 10.2.0.0/24 is used by dev-vpn-vnet to host VirtualNetworkGateway
 	var x, y int
 	rand.Seed(time.Now().UnixNano())
-	// Local Dev clusters are limited to /16 dev-vnet
-	if !c.ci {
-		x, y = 0, 2*rand.Intn(128)
-	} else {
-		x, y = rand.Intn((124))+3, 2*rand.Intn(128)
-	}
+	// Local Dev clusters are limited to /9 dev-vnet
+	x, y = rand.Intn((124))+3, 2*rand.Intn(128)
```

Comment:
I think there is a redundant parenthesis around _124_.

Also, two things to comment here, just my personal opinion 😃 
- First: What about break the line a bit, it would improve readability and we can delete the line _var x, y int_, something like:
```go
secondOctet := rand.Intn(124) + 3
thirdOctet := 2 * rand.Intn(128)

vnetPrefix = fmt.Sprintf("10.%d.%d.0/23", secondOctet, thirdOctet)
masterSubnet = fmt.Sprintf("10.%d.%d.0/24", secondOctet, thirdOctet)
workerSubnet = fmt.Sprintf("10.%d.%d.0/24", secondOctet, thirdOctet+1)
```
- Second: maybe it would be nice to pass the strings with the form"10.%d.%d.0/23" and add some unit tests to ensure the function is behaving as expected, because right now we are assuming something about the Azure environment that can potentially change. Let's add unit tests to express that + pass the parameters to make it extendable. In other words, I think we should not assume anything about the environment and instead pass values to the function.

Reply:
Regarding the parenthesis, I  don't know why I have this tendency to not to touch an existing code it being whatever way it is. I did see the redundant parenthesis, but I just kept it as is.

I agree with your both the opinions and I will change the code and add the unit tests.  However, I want to have a discussion first on the `dev-vnet` vnet in `v4-eastus` rg and to have a timeline about when we will be changing it because if this code gets merged, creation of dev cluster will often fail since the subnet is /16 only.

Reply:
Perfect! The idea bout injecting the parameters in the function and adding unit tests also goes in the direction of being able to make the function work in all cases passing from outside /16 or the corresponding one depending on the region.